compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -92,7 +91,6 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
                         ControlFlow::CONTINUE
                     }
                     Node::Cast(_, _, ty) => {
-                        let ty = ty.subst(tcx, ct.substs);
```

Comment:
I don't think this is correct, you aren't substing this in `AbstractConst::root` like you do for leafs